### PR TITLE
Moving annotations from the reader reducer

### PR DESCRIPTION
**Pull request by tejans24**
connects #2829 

NOTE: #4055 needs to be looked at before this

This is the first part of the PR.
Next part includes the test fixes.

### CODE REVIEW

File: client/app/reader/AnnotationLayer/AnnotationLayerReducer.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      deleteAnnotationModalIsOpenFor: {
+        $set: annotationId
+      }
+    }
+  });
+
+const initialState = {
+  annotations: {},
+  placingAnnotationIconPageCoords: null,
+  ui: {
```
Comment (mdbenjam) [inline client/app/reader/AnnotationLayer/AnnotationLayerReducer.js]: I think the flatter our state the better, do we need this nested state?

File: client/app/reader/AnnotationLayer/AnnotationLayerReducer.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
+import * as Constants from './actionTypes';
+import { update } from '../../util/ReducerUtil';
+import _ from 'lodash';
+import { moveModel } from '../utils';
+
+const openAnnotationDeleteModalFor = (state, annotationId) =>
```
Comment (mdbenjam) [inline client/app/reader/AnnotationLayer/AnnotationLayerReducer.js]: seem like `toggleAnnotationDeleteModalFor` might be a better name.

File: client/app/reader/AnnotationLayer/AnnotationLayerReducer.js (hunk 1 above)
Comment (tejans24) [reply client/app/reader/AnnotationLayer/AnnotationLayerReducer.js]: 👍🏼 I was strongly considering to remove it but I wanted to keep it consistent with other reducers. I'll just flatten it.

**Comment by tejans24** [comment]
UPDATES: I removed the ui and pdf elements in the annotationLayer reducer.

**Pull request merged**